api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -207,12 +207,22 @@ func (mw *azureManagedControlPlaneWebhook) ValidateUpdate(ctx context.Context, o
 		allErrs = append(allErrs, err)
 	}
 
-	if err := webhookutils.ValidateImmutable(
-		field.NewPath("Spec", "DNSPrefix"),
-		m.Spec.DNSPrefix,
-		old.Spec.DNSPrefix,
-	); err != nil {
-		allErrs = append(allErrs, err)
+	if reflect.ValueOf(old.Spec.DNSPrefix).IsZero() && !reflect.ValueOf(m.Spec.DNSPrefix).IsZero() {
```

Comment:
yup that makes perfect sense, thanks! I will update and test this on Monday!

Reply:
After running this it seems to give an error related to not initializing the new spec of DNSPrefix, my initial thought was that it should be defaulted in case of not existing but that error from the last test run makes me think that something else is happening. I initally got the same error when testing that's why I added the `!reflect.ValueOf(m.Spec.DNSPrefix).IsZero()` check for the new DNSPrefix